parsec/core/sync_monitor.py

```diff
@@ -257,8 +261,11 @@ async def tick(self) -> float:
                     # the write access in the future) but pretent it just accured
                     # to avoid a busy sync loop until `read_only` flag is updated.
                     self._local_changes[entry_id] = LocalChange(now)
-                except FSWorkspaceInMaintenance:
-                    # Not the right time for the sync, retry later
+                except (FSWorkspaceInMaintenance, FSBadEncryptionRevision):
+                    # Not the right time for the sync, retry later.
+                    # `FSBadEncryptionRevision` occurs if the reencryption is quick
+                    # enough to start and finish before we process the sharing.reencrypted
+                    # message so we try a sync with the old encryption revision.
```

Comment:
This fix seems OK, but I'm a bit worried that we can no longer tell the difference between an acceptable race condition (accessing the realm while the new encryption key hasn't been received) and bug in the reencryption logic. Is there a way to tell those appart and maybe log the latter?